exchange/bidder.go

```diff
@@ -178,6 +199,64 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, request *openrtb.Bi
 	return seatBid, errs
 }
 
+func addNativeTypes(bid *openrtb.Bid, request *openrtb.BidRequest) (*nativeResponse.Response, []error) {
+	var errs []error
+	var nativeMarkup *nativeResponse.Response
+	if err := json.Unmarshal(json.RawMessage(bid.AdM), &nativeMarkup); err != nil {
+		// Some bidders are returning non-IAB complaiant native markup. In this case Prebid server will not be able to add types. E.g Facebook
+		return nil, errs
+	}
+
+	nativeImp, err := getNativeImpByImpID(bid.ImpID, request)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	var nativePayload nativeRequests.Request
+	if err := json.Unmarshal(json.RawMessage((*nativeImp).Request), &nativePayload); err != nil {
```

Comment:
We have a potential panic here. If `getNativeImpByImpID()` fails to find the imp, then `nativeImp` is nil. Trying to access `nil.Request` will then panic, as you cannot dereference a `nil`. You should probably abort above if `nativeImp` is `nil`. I don't think we need to abort here on `Unmarshal` errors as the worst that should happen is we fail asset lookups and don't do anything.

Reply:
Done